src/bun.js/webcore/response.zig

```diff
@@ -2136,70 +2068,12 @@ pub const Fetch = struct {
-                            if (tls.get(ctx, "ca")) |ca_opt| {
-                                if (ca_opt.isString()) {
-                                    const as_str = ca_opt.getZigString(globalThis);
-                                    ca = as_str.toOwnedSliceZ(allocator) catch @panic("OOM");
-                                }
-
-                                if (ca_opt.isBuffer(ctx.ptr())) {
-                                    if (JSC.Node.StringOrBuffer.fromJS(globalThis, allocator, ca_opt, exception)) |sb| {
-                                        ca = sb.slice();
-                                    }
-                                }
-                            }
-                            if (tls.get(ctx, "cert")) |cert_opt| {
-                                if (cert_opt.isString()) {
-                                    const as_str = cert_opt.getZigString(globalThis);
-                                    cert = as_str.toOwnedSliceZ(allocator) catch @panic("OOM");
-                                }
-
-                                if (cert_opt.isBuffer(ctx.ptr())) {
-                                    if (JSC.Node.StringOrBuffer.fromJS(globalThis, allocator, cert_opt, exception)) |sb| {
-                                        cert = sb.slice();
-                                    }
-                                }
-                            }
-                            if (tls.get(ctx, "key")) |key_opt| {
-                                if (key_opt.isString()) {
-                                    const as_str = key_opt.getZigString(globalThis);
-                                    key = as_str.toOwnedSliceZ(allocator) catch @panic("OOM");
-                                }
-
-                                if (key_opt.isBuffer(ctx.ptr())) {
-                                    if (JSC.Node.StringOrBuffer.fromJS(globalThis, allocator, key_opt, exception)) |sb| {
-                                        key = sb.slice();
-                                    }
-                                }
+                            if (ssl_config) |*conf| {
+                                conf.deinit();
+                                ssl_config = null;
                             }
-                            if (tls.get(ctx, "passphrase")) |passphrase_opt| {
-                                if (passphrase_opt.isString()) {
-                                    const as_str = passphrase_opt.getZigString(globalThis);
-                                    passphrase = as_str.toOwnedSliceZ(allocator) catch @panic("OOM");
-                                }
+                            if (SSLConfig.inJS(globalThis, tls, exception)) |config| {
+                                ssl_config = config;
                             }
                             if (!tls.isEmptyOrUndefinedOrNull() and tls.isObject()) {
                                 if (tls.get(ctx, "rejectUnauthorized")) |reject| {
```

Comment:
I think `rejectUnauthorized` should be removed, since it's already inside ssl_config too.

Reply:
actually no this needs to stay since the bool inside the config is not passed to the HttpClient property, just taking the value without checking the existance of the property is wrong too.